docs/v3/core/index.rst

```diff
@@ -1599,46 +1471,266 @@ Storage transformers may be stacked to combine different functionalities:
+MUST be a plain string which conforms to :ref:`extension name <extension-naming>`.
+Optionally, the member ``configuration`` MAY be present but if so MUST be
+an object.
+
+For example::
+
+    {
+        "name": "<name>",        # conformant name
+        "configuration": { ... } # optional object
+    }
+
+.. _extension-definition-short-hand-name:
+
+Short-hand names
+^^^^^^^^^^^^^^^^
+
+Instead of extension objects, short-hand names MAY be used if no
+configuration metadata is required. They are equivalent to extension
+objects with just a `name` key.
+
+.. _extension-definition-must-understand:
+
+`must_understand`
+^^^^^^^^^^^^^^^^^
+
+An extension object is interpreted to have an implicit field `must_understand` set to
+`True`, unless otherwise stated. An extension object MAY explicitly set `must_understand=False` if
+implementations can ignore its presence.
+
+An implementation MUST fail to open Zarr groups or arrays if any
+metadata fields are present which (a) the
+implementation does not recognize and (b) are not explicitly
+set to ``"must_understand": false``.
+
+`must_understand=False` is not supported for the following extension points:
+data type, chunk grid, and chunk key encoding.
+
+Use of `must_understand=False` to add top-level keys is discouraged in favor
+of the explicit use of :ref:`extension-points`.
+
+.. _extension-naming:
+
+Extension naming
+----------------
+
+The `name` field of an extension is an identifier taking one of two forms:
+**registered names** (as simple strings) and **unregistered names** (as URIs).
+
+Implementations SHOULD be able to resolve multiple names to the same
+implementation to support unregistered names which are subsequently registered.
+
+.. _extension-naming-registered-names:
+
+Registered names
+^^^^^^^^^^^^^^^^
+
+Registered names consist of a single string that is unique within the Zarr ecosystem, with no prefix.
+Registered names are intended for well-known extensions aimed at broad adoption and maximum interoperability.
+Registered names are unique and immutable.
+Registered names MUST start with one lower case letter a-z and then be followed
+by only lower case letters a-z, numerals 0-9, underscores, dots and dashes.
+
+Registered names MUST be assigned within a central repository, `zarr-extensions`_
+a Github repository, where extensions and their specification are listed.
+The Zarr Steering Council or by delegation a
+maintainer team reserves the right to refuse name assignment at its own
+discretion.
+
+- **Accepted regex:** ``^[a-z][a-z0-9-_.]+$``
+- **Valid examples:**
+  - ``zstd``
+  - ``numcodecs.adler32``
+- **Invalid examples:**
+  - ``foo/bar``
+
+.. _extension-naming-unregistered-names:
+
+Unregistered names
+^^^^^^^^^^^^^^^^^^
+
+Unregistered names are intended for private extensions and for experimental and development purposes.
+
+Unregistered names are not centrally managed and MAY be used by any extension without coordination.
```

Comment:
if these two statements are true, then why are we imposing any requirements on the structure of unregistered?